src/Functions/FunctionStrncmp.cpp

```diff
@@ -0,0 +1,445 @@
+
+        auto first_offset = arguments[1].column->getUInt(0);
+        auto second_offset = arguments[3].column->getUInt(0);
+        auto bytes = arguments[4].column->getUInt(0);
+
+        auto & result = compare_result_column->getData();
+
+        if (first_string && second_const_string)
+        {
+            executeStringWithConstString<true>(
+                *first_string, first_offset, String(second_const_string->getDataAt(0)), second_offset, bytes, result);
+        }
+        else if (first_string && second_string)
+        {
+            executeStringWithString(*first_string, first_offset, *second_string, second_offset, bytes, result);
+        }
+        else if (first_fixed_string && second_fixed_string)
+        {
+            executeFixedStringWithFixedString(*first_fixed_string, first_offset, *second_fixed_string, second_offset, bytes, result);
+        }
+        else if (first_fixed_string && second_string)
+        {
+            executeFixedStringWithString<true>(*first_fixed_string, first_offset, *second_string, second_offset, bytes, result);
+        }
+        else if (first_string && second_fixed_string)
+        {
+            executeFixedStringWithString<false>(*second_fixed_string, second_offset, *first_string, first_offset, bytes, result);
+        }
+        else if (first_const_string && second_string)
+        {
+            executeStringWithConstString<false>(
+                *second_string, second_offset, String(first_const_string->getDataAt(0)), first_offset, bytes, result);
+        }
+        else if (first_fixed_string && second_const_string)
+        {
+            executeFixedStringWithConst<true>(
+                *first_fixed_string, first_offset, String(second_const_string->getDataAt(0)), second_offset, bytes, result);
+        }
+        else if (second_fixed_string && first_const_string)
+        {
+            executeFixedStringWithConst<false>(
+                *second_fixed_string, second_offset, String(first_const_string->getDataAt(0)), first_offset, bytes, result);
+        }
+        else if (first_const_string && second_const_string)
+        {
+            executeConstWithConst(
+                String(first_const_string->getDataAt(0)),
+                first_offset,
+                String(second_const_string->getDataAt(0)),
+                second_offset,
+                bytes,
+                arguments[0].column->size(),
+                result);
+        }
+        else
+        {
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal columns {} and {} of arguments of function {}",
+                first_str_col->getName(),
+                second_str_col->getName(),
+                getName());
+        }
+        return std::move(compare_result_column);
+    }
+
+private:
+    void checkArguments(const ColumnsWithTypeAndName & args) const
+    {
+        if (args.size() != 5)
```

Comment:
Please remove function `checkArguments()`. Instead, please use `FunctionArgumentDescriptor` and `validateFunctionArguments` from src/Functions/FunctionHelpers.h to do the validation in `getReturnTypeImpl` (l. 42).